extensions/shared/submit-button.js

```diff
@@ -1,138 +1,191 @@
+		[ textColor.class ]: textColor.class,
+		[ gradientClass ]: gradientClass,
+		'no-border-radius': borderRadius === 0,
+		[ fontSize.class ]: fontSize.class,
+	} );
 
-	return {
-		fallbackBackgroundColor: backgroundColorValue || fallbackBackgroundColor,
-		fallbackTextColor: textColorValue || fallbackTextColor,
+	const styles = {
+		...( ! backgroundColor.color && gradientValue
+			? { background: gradientValue }
+			: { backgroundColor: backgroundColor.color } ),
+		color: textColor.color,
+		borderRadius: borderRadius ? borderRadius + 'px' : undefined,
+		fontSize: fontSize.size ? fontSize.size + 'px' : undefined,
 	};
-} );
-
-class SubmitButton extends Component {
-	componentDidUpdate( prevProps ) {
-		if (
-			! isEqual( this.props.textButtonColor, prevProps.textButtonColor ) ||
-			! isEqual( this.props.backgroundButtonColor, prevProps.backgroundButtonColor )
-		) {
-			const buttonClasses = this.getButtonClasses();
-			this.props.setAttributes( { submitButtonClasses: buttonClasses } );
-		}
-	}
-	getButtonClasses() {
-		const { textButtonColor, backgroundButtonColor } = this.props;
-		const textClass = get( textButtonColor, 'class' );
-		const backgroundClass = get( backgroundButtonColor, 'class' );
-		return classnames( 'wp-block-button__link', {
-			'has-text-color': textButtonColor.color,
-			[ textClass ]: textClass,
-			'has-background': backgroundButtonColor.color,
-			[ backgroundClass ]: backgroundClass,
-		} );
-	}
-	render() {
-		const {
-			attributes,
-			backgroundButtonColor,
-			textButtonColor,
-			fallbackBackgroundColor,
-			fallbackTextColor,
-			setAttributes,
-			setBackgroundButtonColor,
-			setTextButtonColor,
-		} = this.props;
 
-		const backgroundColor = backgroundButtonColor.color || fallbackBackgroundColor;
-		const color = textButtonColor.color || fallbackTextColor;
-		const buttonStyle = { border: 'none', backgroundColor, color };
-		const buttonClasses = this.getButtonClasses();
+	return (
+		<div className="wp-block-button">
+			<RichText
+				allowedFormats={ [] }
+				className={ classes }
+				onChange={ value => setAttributes( { submitButtonText: value } ) }
+				placeholder={ __( 'Add text…', 'jetpack' ) }
+				style={ styles }
+				value={ submitButtonText }
+				withoutInteractiveFormatting
+			/>
 
-		return (
-			<Fragment>
-				<div className="wp-block-button jetpack-submit-button">
-					<RichText
-						placeholder={ __( 'Add text…', 'jetpack' ) }
-						value={ attributes.submitButtonText }
-						onChange={ nextValue => setAttributes( { submitButtonText: nextValue } ) }
-						className={ buttonClasses }
-						style={ buttonStyle }
-						keepPlaceholderOnFocus
-						allowedFormats={ [] }
-					/>
-				</div>
-				<InspectorControls>
+			<InspectorControls>
+				<PanelBody title={ __( 'Button Text Settings' ) } initialOpen={ false }>
+					<FontSizePicker value={ fontSize.size } onChange={ setFontSize } />
+				</PanelBody>
+				{ isGradientAvailable && (
+					<PanelColorGradientSettings
```

Comment:
Should the color also apply to the input border?

Reply:
Based on the patterns in the issue, I think the border color needs to be separate.